addons/payment/models/payment_acquirer.py

```diff
@@ -535,6 +539,13 @@ def get_next_reference(self, reference):
             ref_suffix += 1
         return reference
 
+    def _generate_callback_hash(self):
+        self.ensure_one()
+        secret = self.env['ir.config_parameter'].sudo().get_param('database.secret')
+        token = '%s%s%s' % (self.callback_ir_model_id.model, self.callback_res_id, self.callback_method)
+        hm = hmac.new(str(secret), token, hashlib.sha1).hexdigest()
```

Comment:
I'd rather we [don't use SHA1](http://arstechnica.com/security/2015/10/sha1-crypto-algorithm-securing-internet-could-break-by-years-end/).

A performance test for 1M hmac generation with SHA256 indicates a 3-7% difference in cpu time, I don't think we'll have any performance problem there, please use SHA256 instead of SHA1.